pkg/client/client.go

```diff
@@ -776,12 +775,9 @@ func (c *Client) CreateOrUpdateDeployment(ctx context.Context, dep *appsv1.Deplo
 			if err != nil {
 				return errors.Wrap(err, "deleting Deployment object failed")
 			}
-			retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				return c.CreateDeployment(ctx, required)
-			})
-
-			if retryErr != nil {
-				return errors.Wrap(retryErr, "creating Deployment object failed after update failed")
+			err = c.CreateDeployment(ctx, required)
```

Comment:
@slashpai May I know why `RetryOnConflict` is no longer useful?

Reply:
The exact background can be found here: https://github.com/openshift/cluster-monitoring-operator/pull/1333#issuecomment-902802506